### PR TITLE
Render body according to its request content type (#261)

### LINKED ISSUE #261: Body in the report should be rendered according to it's request content type
Today the Body is showing the representation of python's byte string, which in the report should be agnostic of language, but based on it's request content type.

For example:

- A request of application/json can render the body as json in the report and make use of the same tool that is used at the Content field
- A request of binary content type should just show a placeholder text like: "Binary content"
- A request of text content type should show as a text without python's representation

### PR DESCRIPTION
**Pull request by beatorizu**
Resolves #261

### CODE REVIEW

File: scanapi/template_render.py [hunk 1]
```diff
@@ -18,3 +19,13 @@ def _loader(is_external):
...
 
     return PackageLoader("scanapi", "templates")
+
+
+def render_body(request):
+    """ Render body according to its request content type """
+    content_type = request.headers.get("Content-Type")
+    body = request.body
+    if content_type in ["application/json", "text/plain"]:
+        return body.decode("UTF-8")
```
Comment (marcuxyz) [inline scanapi/template_render.py:29]: since python 3 not necessary set utf-8 in decode

File: scanapi/template_render.py [hunk 2]
```diff
@@ -18,3 +19,13 @@ def _loader(is_external):
         return FileSystemLoader(searchpath="./")
 
     return PackageLoader("scanapi", "templates")
+
+
+def render_body(request):
+    """ Render body according to its request content type """
+    content_type = request.headers.get("Content-Type")
+    body = request.body
```
Comment (marcuxyz) [inline scanapi/template_render.py:27]: you can returns request.body.decode() directly. Cause, you use only once.

**Review by marcuxyz — CHANGES_REQUESTED** [review]

File: scanapi/template_render.py (hunk 1 above)
Comment (beatorizu) [reply scanapi/template_render.py:29]: Yep, decode will use the default string encoding

**Review by marcuxyz — APPROVED** [review]

File: scanapi/template_render.py (hunk 1 above)
Comment (marcuxyz) [reply scanapi/template_render.py]: Thanks !

File: scanapi/template_render.py [hunk 3]
```diff
@@ -20,3 +21,12 @@ def _loader(is_external):
         return FileSystemLoader(searchpath="./")
 
     return PackageLoader("scanapi", "templates")
+
+
+def render_body(request):
```
Comment (camilamaia) [inline scanapi/template_render.py:26]: Would it be possible to have tests for this? The test file for this part is https://github.com/scanapi/scanapi/blob/master/tests/unit/test_template_render.py

Here you can find a brief guideline to help you with the tests: https://github.com/scanapi/scanapi/wiki/Run-ScanAPI-Locally#tests

**Review by camilamaia — COMMENTED** [review]
Hi @beatorizu, I am awfully sorry for the **long** delay answering you here. The last month was too busy and I didn't have time to follow all the PRs here.

The PR looks great, btw! I've just left one comment about tests. Please, let me know if you need any help 👍 And thanks for the contribution!

File: scanapi/templates/report.html [hunk 4]
```diff
@@ -449,7 +449,8 @@ <h3>Request</h3>
                     {% if ( request.body and request.body.decode("UTF-8") != "{}" ) %}
                     <details>
                         <summary>Body</summary>
-                        <p>{{ request.body }}</p>
+                        <p class="json-body">{{ request|render_body }}</p>
+                        <button class="copy__snippet">&#10063;</button>
```
Comment (camilamaia) [inline scanapi/templates/report.html:453]: 👏👏👏

**Comment by beatorizu** [comment]
I changed my setup recentely, so I will set the environment and add the test :)

**Review by camilamaia — APPROVED** [review]
👏👏👏

**Pull request merged**